lib/annotate/annotate_routes.rb

```diff
@@ -30,7 +30,13 @@ def do_annotations(options = {})
         new_content = annotate_routes(HeaderGenerator.generate(options), content, header_position, options)
         new_text = new_content.join("\n")
 
-        if rewrite_contents(existing_text, new_text)
+        if options[:frozen]
+          if needs_rewrite_contents?(existing_text, new_text)
+            abort "annotate error. #{routes_file} needs to be updated, but annotate was run with `--frozen`."
+          else
+            puts "#{routes_file} was not changed."
+          end
+        elsif rewrite_contents(existing_text, new_text)
           puts "#{routes_file} was annotated."
         else
           puts "#{routes_file} was not changed."
```

Comment:
I feel this can be somewhat improved to remove the code duplication and nested ifs.  What about something like this?

```ruby
# don't feel like a separate function is needed
content_changed = (existing_text.strip != new_text.strip)  

if !content_changed # you can flip this condition around if you want
  puts "#{routes_file} was not changed."
else
  abort "annotate error. #{routes_file} needs to be updated, but annotate was run with `--frozen`." if options[:frozen]
  rewrite_contents(existing_text, new_text)  
end
```

Reply:
Now that I'm looking at the code closer, maybe we should move all this inside the `rewrite_contents` function.

Reply:
Keep this code as is:
```ruby
if rewrite_contents(existing_text, new_text)
  puts "#{routes_file} was annotated."
else
   puts "#{routes_file} was not changed."
end
```

Then change `#rewrite_contents` to:
```ruby
def rewrite_contents(existing_text, new_text)
  content_changed = (existing_text.strip != new_text.strip)

  # Rubocop may prefer "abort ... if"
  if content_changed && options[:frozen] 
    abort "annotate error. #{routes_file} needs to be updated, but annotate was run with `--frozen`." 
  end

  File.open(routes_file, 'wb') { |f| f.puts(new_text) }
  content_changed
end
```